app/actions/websocket/categories.ts

```diff
@@ -0,0 +1,62 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import {getCurrentTeamId} from '@mm-redux/selectors/entities/teams';
+import {ActionResult, DispatchFunc, GetStateFunc} from '@mm-redux/types/actions';
+import {WebSocketMessage} from '@mm-redux/types/websocket';
+import {fetchMyCategories, receivedCategoryOrder} from '@mm-redux/actions/channel_categories';
+
+export function handleSidebarCategoryCreated(msg: WebSocketMessage) {
+    return async (dispatch: DispatchFunc, getState: GetStateFunc): Promise<ActionResult> => {
+        const state = getState();
+        const currentTeamId = getCurrentTeamId(state);
+
+        if (msg.broadcast.team_id !== currentTeamId) {
+            // The new category will be loaded when we switch teams.
+            return {data: false};
+        }
+
+        // Fetch all categories, including ones that weren't explicitly updated, in case any other categories had channels
+        // moved out of them.
+        dispatch(fetchMyCategories(msg.broadcast.team_id));
```

Comment:
Is there a better way to do this? It seems over reach to have to request everything again

Reply:
Not sure - I'll dig into the websocket data and see if it passes through just whats changed for the categories; and perhaps just trigger updates to the state from there.

WebApp just calls the server - though I'm not sure which is less of a penalty; diffing the change and then updating state or just making this call and updating state.